orchestration/dagster_orchestration/hca_orchestration/tests/test_pipelines.py

```diff
@@ -1,37 +1,71 @@
+import os
+import pytest
 import unittest
 from unittest.mock import patch
+import uuid
+from typing import Any
 
-import os
-from dagster import execute_pipeline, file_relative_path
+
+from dagster import execute_pipeline, file_relative_path, PipelineDefinition, PipelineExecutionResult
 from dagster.utils import load_yaml_from_globs
+from dagster.utils.merger import deep_merge_dicts
 from hca_orchestration.pipelines import stage_data, validate_egress
+from hca_manage.diff_dirs import diff_dirs
+from hca_orchestration.support.typing import DagsterConfigDict
 
 
-def config_path(relative_path):
-    return file_relative_path(
+def config_path(relative_path: str) -> str:
+    path: str = file_relative_path(
         __file__, os.path.join("../environments/", relative_path)
     )
+    return path
 
 
 class PipelinesTestCase(unittest.TestCase):
-    def run_pipeline(self, pipeline, config_name, *execution_args, **execution_kwargs):
+    def run_pipeline(self, pipeline: PipelineDefinition, config_name: str, extra_config: dict[str, Any] = {},
```

Comment:
once you've pulled in my typing changes you should be able to extend `DagsterConfigDict` in `hca_orchestration.support.typing` to produce a more restrictive type for pipeline config dicts